nexus/server.go

```diff
@@ -26,17 +26,66 @@ type handlerCtx struct {
 	links []Link
 }
 
+// WithHandlerContext returns a new context from a given context setting it up for being used for handler methods.
+// Meant to be used by frameworks, not directly by applications.
+//
+// NOTE: Experimental
+func WithHandlerContext(ctx context.Context) context.Context {
+	return context.WithValue(ctx, handlerCtxKey, &handlerCtx{})
+}
+
+// IsHandlerContext returns true if the given context is a handler context where [AddHandlerLinks] and [HandlerLinks]
+// can be called. It will only return true when called from an [Operation] handler Start method or from a [Handler]
+// StartOperation method.
+//
+// NOTE: Experimental
+func IsHandlerContext(ctx context.Context) bool {
+	return ctx.Value(handlerCtxKey) != nil
+}
+
+// HandlerLinks retrieves the attached links on the given handler context. The returned slice should not be mutated.
+// The context provided must be the context passed to the handler or this method will panic, [IsHandlerContext] can be
+// used to verify the context is valid.
+//
+// NOTE: Experimental
+func HandlerLinks(ctx context.Context) []Link {
+	hctx := ctx.Value(handlerCtxKey).(*handlerCtx)
+	hctx.mu.Lock()
+	links := hctx.links
+	hctx.mu.Unlock()
+	cpy := make([]Link, len(links))
+	copy(cpy, links)
```

Comment:
```suggestion
	cpy := make([]Link, len(links))
	copy(cpy, links)
	hctx.mu.Unlock()
```

The lock needs to be held while it is accessed

Reply:
Why? We always assign to `h.links`, we never mutate it directly.

Reply:
It's a common misunderstanding in Go that `append` doesn't mutate the first param

Reply:
Thanks, I wasn't aware.